distribution/tools/keystore-cli/src/test/java/org/elasticsearch/bootstrap/BootstrapTests.java

```diff
@@ -66,4 +78,43 @@ public void testLoadSecureSettings() throws Exception {
             assertTrue(Files.exists(configPath.resolve("elasticsearch.keystore")));
         }
     }
+
+    public void testReadCharsFromStdin() throws Exception {
+        assertPassphraseRead("\n", "");
+        assertPassphraseRead("\r", "");
+        assertPassphraseRead("\r\n", "");
+
+        assertPassphraseRead("hello", "hello");
+        assertPassphraseRead("hello\n", "hello");
+        assertPassphraseRead("hello\r", "hello");
+        assertPassphraseRead("hello\r\n", "hello");
+
+        assertPassphraseRead("hellohello", "hellohello");
+        assertPassphraseRead("hellohello\n", "hellohello");
+        assertPassphraseRead("hellohello\r", "hellohello");
+        assertPassphraseRead("hellohello\r\n", "hellohello");
+
+        assertPassphraseRead("hello\nhi\n", "hello");
+        assertPassphraseRead("hello\rhi\r", "hello");
+        assertPassphraseRead("hello\r\nhi\r\n", "hello");
```

Comment:
what about negative assertions, ie passphrase too long?

Reply:
I cover that in a test below, but I need to adjust the names make it more clear.